### PR TITLE
4.1.x - mentor can track multiple answers simultaneously with option to cancel

### PR DESCRIPTION
**Pull request by aaronshiel**
A completed job is dismissed by either:
- Visiting its respective page or
- Pressing the dismiss x button

### CODE REVIEW

File: client/src/components/record/uploading-widget.tsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    (upload) =>
+      upload.uploadStatus !== UploadStatus.DONE &&
+      upload.uploadStatus !== UploadStatus.UPLOAD_FAILED
+  );
+  const height = 250;
+  const width = 350;
+
+  //the IDx of an answer corresponds to its position within the answers array
+  function retrieveAnswerIDx(id: string) {
+    let i = 0;
```
Comment (beatthat) [inline client/src/components/record/uploading-widget.tsx]: why is this `i` scoped outside the for loop? Shoudn't be this?

```typescript
for (const i = 0; i < answers?.length; i++) {
```

**Review by beatthat — CHANGES_REQUESTED** [review]
few change requests inline. the most important one is the one about not using a form to post the cancel json...do we need that for some reason?

File: client/src/api.ts [hunk 2]
```diff
@@ -625,14 +627,31 @@ export async function uploadVideo(
...
+  mentorId: string,
+  question: Question,
+  taskId: string
+): Promise<CancelJob> {
+  const data = new FormData();
+  data.append(
+    "body",
+    JSON.stringify({ mentor: mentorId, question: question._id, task: taskId })
+  );
+  const result = await uploadRequest.post("/answer/cancel", data);
```
Comment (beatthat) [inline client/src/api.ts]: I think this should just be:

```typescript
const result = await uploadRequest.post(
  "/answer/cancel", 
  { mentor: mentorId, question: question._id, task: taskId }
);
```

...or if not, why not?

**Review by beatthat — APPROVED** [review]

File: client/src/api.ts (hunk 2 above)
Comment (kycarr) [reply client/src/api.ts]: it is

**Pull request merged**